lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -0,0 +1,316 @@
+module Msf::Payload::Adapter::Fetch
+
+  def initialize(*args)
+    super
+    register_options(
+      [
+        Msf::OptEnum.new('FETCH_COMMAND', [true, 'Command to fetch payload', 'CURL', %w{ WGET CURL FTP TNFTP TFTP Certutil Psh_InvokeWebRequest }]),
+        Msf::OptPort.new('FETCH_SRVPORT', [true, 'Local port to use for serving payload', 8080]),
+        Msf::OptAddressLocal.new('FETCH_SRVHOST', [ true, 'Local IP to use for serving payload', "0.0.0.0"]),
```

Comment:
It looks like there's a missing check for this. `0.0.0.0` will never be usable because it can't be connected to.

You probably want a check like this somewhere which will validate that if it's an IP address that it's not the all-interfaces.

```
Rex::Socket.is_ip_addr?(datastore['FETCH_SRVHOST']) && Rex::Socket.addr_atoi(datastore['FETCH_SRVHOST']) == 0
```

Even when `FetchListenerBindAddress` is in use, this would still need to be a routeable address.

Reply:
Yup; the code for it is there, but the call to it was removed during fixes.  I re-added it.